scalding-core/src/main/scala/com/twitter/scalding/ArgHelp.scala

```diff
@@ -0,0 +1,106 @@
+package com.twitter.scalding
+
+sealed abstract class DescribedArg {
+  def key: String
+  def description: String
+}
+
+case class RequiredArg(key: String, description: String) extends DescribedArg
+case class OptionalArg(key: String, description: String) extends DescribedArg
+case class ListArg(key: String, description: String) extends DescribedArg
+case class BooleanArg(key: String, description: String) extends DescribedArg
+
+trait ArgHelper {
+  /**
+   * Similar to describe but validate all args are described
+   * @param describedArgs List of Argument Descriptions
+   * @param ex Input Execution
+   * @return Output Execution
+   */
+  def validatedDescribe(describedArgs: Seq[DescribedArg], ex: Execution[Unit]): Execution[Unit] = {
```

Comment:
Any reason not to put a type T on the Executions so you can pass through your state here?